tests/gdb-tests/tests/test_command_stepuntilasm.py

```diff
@@ -0,0 +1,29 @@
+import gdb
+import pytest
+
+import pwndbg.gdblib
+import tests
+
+STEPUNTILASM_X64_BINARY = tests.binaries.get("stepuntilasm_x64.out")
+
+
+@pytest.mark.parametrize("binary", [STEPUNTILASM_X64_BINARY], ids=["x86-64"])
+def test_command_untilasm_x64(start_binary, binary):
+    """
+    Tests the chain for a non-nested linked list
+    """
+
+    start_binary(binary)
+    gdb.execute("break break_here")
+    gdb.execute("run")
+
+    run_and_verify("stop1", "nop")
+    run_and_verify("stop2", "xor rax, rax")
+    run_and_verify("stop3", "mov qword ptr [rax], 0x20")
+    run_and_verify("stop4", "mov dword ptr [rax+4], 0x20")
```

Comment:
How does this work with multiple whitespaces? What will happen if the user types in `stepuntilasm xor rax,rax` or `stepuntilasm xor  rax , rax`?

Can we maybe handle this somehow? We could maybe take the whole instruction as a string and split on whitespaces and commas?

(Would it make things notably slower? If so, maybe that's not a good idea to handle it)

Reply:
Oh, we already do! All whitespace other than a single one separating the mnemonic from the operands is ignored. Which means `xor   rax,rax` is `xor     rax ,rax` is `xor  rax , rax` is `xor rax, rax`. Case differences are also ignored. 

This means you can match on `mov qwordptr[rax],0x20` too, but since these combined keywords aren't themselves different valid keywords, it should be fine.

In terms of performance, while it does make it slower, it's still fairly insignificant compared to the time it takes to do the back and forth from GDB in my machine. If you want some numbers for the procedure this patch uses to ignore whitespace and case differences, though, here they are:
```py
def casefold_whitespace():
    for _ in range(100000):
        ops = "".join("rax, rax".split()).casefold()

        tmp = []
        for op in ["rax", ",       ", "     rax"]:
            tmp.extend(op.split())
        op_str = "".join(tmp).casefold()

        a = ops == op_str
```

```
         1100004 function calls in 0.264 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    0.264    0.264 <string>:1(<module>)
        1    0.170    0.170    0.264    0.264 a.py:1(casefold_whitespace)
        1    0.000    0.000    0.264    0.264 {built-in method builtins.exec}
   200000    0.015    0.000    0.015    0.000 {method 'casefold' of 'str' objects}
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
   300000    0.021    0.000    0.021    0.000 {method 'extend' of 'list' objects}
   200000    0.018    0.000    0.018    0.000 {method 'join' of 'str' objects}
   400000    0.040    0.000    0.040    0.000 {method 'split' of 'str' objects}
```